src/ensure_buffer.ts

```diff
@@ -11,6 +11,10 @@ import { isAnyArrayBuffer } from './parser/utils';
  * @throws BSONTypeError If anything other than a Buffer or Uint8Array is passed in
  */
 export function ensureBuffer(potentialBuffer: Buffer | ArrayBufferView | ArrayBuffer): Buffer {
+  if (potentialBuffer instanceof Buffer) {
```

Comment:
It might be worth leaving a comment here about why this extra check is added :)

Reply:
done!